packages/launchpad/src/migration/fragments/HighlightedFile.vue

```diff
@@ -1,51 +1,29 @@
 <template>
-  <code class="text-14px leading-24px"><template
-    v-for="(part, index) in fileParts"
-    :key="index"
-  ><span
-    v-if="part.highlight"
-    :class="props.highlightClass"
-  >{{ part.text }}</span><template
-    v-else
-  >{{ part.text }}</template></template></code>
+  <code class="text-14px leading-24px">
+    <template
+      v-for="(part, index) in parts"
+      :key="index"
+    >
+      <span
+        v-if="part.highlight"
+        :class="props.highlightClass"
+      >
+        {{ part.text }}
+      </span>
+
+      <template v-else>
+        {{ part.text }}
+      </template>
+    </template>
+  </code>
 </template>
 
 <script lang="ts" setup>
-import { computed } from 'vue'
+import type { FilePart } from '../../utils/stringToRegexp'
 
 const props = defineProps<{
-  file: string
-  highlightRegExp: RegExp,
+  parts: readonly FilePart[]
   highlightClass: string
 }>()
 
-const fileParts = computed(() => {
-  let result:RegExpExecArray | null
-  let start:number = 0
-  const allParts: { text:string, highlight:boolean }[] = []
-
-  while ((result = props.highlightRegExp.exec(props.file))) {
```

Comment:
@elevatebart nuked this, better just to use the built-in feature of regular expressions called "named groups". EG

```
`cypress\/(?<dir>${defaultFolder})\/.*?(?<ext>[._-]?[s|S]pec.|[.])(?=[j|t]s[x]?)`
```

```
(?<name-of-group>)
You can name the group and do the replace w/o while/slice/index hacks. Whole thing is just this now: https://github.com/cypress-io/cypress/pull/19807/files#diff-0075775a5a7e9fabfcc8e083c6d37c56439b7055d9a0ec04af8459cd2f68e6deR10-R26